src/Dapr.Messaging/PublishSubscribe/Extensions/PublishSubscribeServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,104 @@
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Dapr.Messaging.PublishSubscribe.Extensions;
+
+/// <summary>
+/// Contains extension methods for using Dapr Publish/Subscribe with dependency injection.
+/// </summary>
+public static class PublishSubscribeServiceCollectionExtensions
+{
+    /// <summary>
+    /// Adds Dapr Publish/Subscribe support to the service collection.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/>.</param>
+    /// <returns></returns>
+    public static IServiceCollection AddDaprPubSubClient(this IServiceCollection services)
+    {
+        ArgumentNullException.ThrowIfNull(services, nameof(services));
+
+        //Register the IHttpClientFactory implementation
+        services.AddHttpClient();
+
+        services.TryAddSingleton(serviceProvider =>
+        {
+            var httpClientFactory = serviceProvider.GetRequiredService<IHttpClientFactory>();
+
+            var builder = new DaprPublishSubscribeClientBuilder();
+            builder.UseHttpClientFactory(httpClientFactory);
+
+            return builder.Build();
+        });
+
+        return services;
+    }
+
+    /// <summary>
+    /// Adds Dapr Publish/Subscribe support to the service collection.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/>.</param>
+    /// <param name="configure">Optionally allows greater configuration of the <see cref="DaprPublishSubscribeClientBuilder"/>.</param>
+    /// <returns></returns>
+    public static IServiceCollection AddDaprPubSubClient(this IServiceCollection services, Action<DaprPublishSubscribeClientBuilder>? configure)
+    {
+        ArgumentNullException.ThrowIfNull(services, nameof(services));
+
+        //Register the IHttpClientFactory implementation
+        services.AddHttpClient();
+
+        services.TryAddSingleton(serviceProvider =>
+        {
+            var httpClientFactory = serviceProvider.GetRequiredService<IHttpClientFactory>();
+
+            var builder = new DaprPublishSubscribeClientBuilder();
+            builder.UseHttpClientFactory(httpClientFactory);
+
+            configure?.Invoke(builder);
+
+            return builder.Build();
+        });
+
+        return services;
+    }
+
+    /// <summary>
+    /// Adds Dapr Publish/Subscribe support to the service collection.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/>.</param>
+    /// <param name="configure">Optionally allows greater configuration of the <see cref="DaprPublishSubscribeClient"/> using injected services.</param>
+    /// <returns></returns>
+    public static IServiceCollection AddDaprPubSubClient(this IServiceCollection services, Action<IServiceProvider, DaprPublishSubscribeClientBuilder>? configure)
```

Comment:
Could the first two overloads ultimately just call into this method?

Reply:
Definitely. Done.